crates/storage-query-datafusion/src/remote_query_scanner_manager.rs

```diff
@@ -191,11 +181,11 @@ impl ScanPartition for RemotePartitionsScanner {
         range: RangeInclusive<PartitionKey>,
         projection: SchemaRef,
     ) -> anyhow::Result<SendableRecordBatchStream> {
-        match self
-            .manager
-            .get_partition_target_node(&self.table_name, partition_id)?
-        {
-            PartitionLocation::Local { scanner } => {
+        match self.manager.get_partition_target_node(partition_id)? {
+            PartitionLocation::Local => {
+                let scanner = self.manager.local_partition_scanner(&self.table_name).ok_or_else(
+                    ||anyhow!("was expecting a local partition to be present on this node. It could be that this partition is being opened right now.")
```

Comment:
Nit:

```suggestion
                    || anyhow!("was expecting a local partition to be present on this node. It could be that this partition is being opened right now.")
```